meinberlin/apps/projects/helpers.py

```diff
@@ -0,0 +1,40 @@
+from django.http import HttpResponse
+from django.shortcuts import render
+from django.db import connection
+
+def public_projects_exec_raw_query():
+	query = """CREATE OR REPLACE VIEW public_project_view AS
+        SELECT 
+            project.*, 
+            district.*, 
+            organisation.*, 
+            moderator.*, 
+            plan.*, 
+            initiators.*, 
+            phases.*
+        FROM 
+            a4projects_project project
+        LEFT JOIN 
+            a4administrative_districts_administrativedistrict district ON project.administrative_district_id = district.id
+        LEFT JOIN 
+            meinberlin_organisations_organisation organisation ON project.organisation_id = organisation.id
+        LEFT JOIN 
+            a4projects_project_moderators moderator ON project.id = moderator.project_id
+        LEFT JOIN 
+            meinberlin_plans_plan plan ON project.id = moderator.project_id
+        LEFT JOIN 
+            meinberlin_organisations_organisation_initiators initiators ON organisation.id = initiators.organisation_id
+        LEFT JOIN 
+            a4modules_module module ON project.id = module.project_id
+        LEFT JOIN 
+            a4phases_phase phases ON module.id = phases.module_id
```

Comment:
Not sure if those are the correct query to get the phases, any ideas welcome. need to research more.

Reply:
hmm, as this is already a rather complex query I'm not super convinced this is the easier way to go (i.e. having to maintain a raw sql query if we change anything on the models vs the ORM just giving us an error) compared to moving the query and the filtering into functions we can call from the cache task. What are your thoughts, I haven't worked a lot with raw sql in django before, so this seems way more complicated to me :/